src/process_utility.c

```diff
@@ -51,7 +52,31 @@ timescaledb_ProcessUtility(Node *parsetree,
 		return;
 	}
 
-	/* We don't support renaming hypertables yet so we need to block it */
+	/* Change the schema of hyper table */
+	if (IsA(parsetree, AlterObjectSchemaStmt))
+	{
+		AlterObjectSchemaStmt *alterstmt = (AlterObjectSchemaStmt *) parsetree;
+		Oid			relId = RangeVarGetRelid(alterstmt->relation, NoLock, true);
+
+		if (OidIsValid(relId))
+		{
+			Cache	   *hcache = hypertable_cache_pin();
+			Hypertable *hentry = hypertable_cache_get_entry(hcache, relId);
+
+			if (hentry != NULL && alterstmt->objectType == OBJECT_TABLE) {
+				executor_level_enter();
+				spi_hypertable_rename(hentry,
+									  alterstmt->newschema,
+									  NameStr(hentry->fd.table_name));
+				executor_level_exit();
+			}
+			cache_release(hcache);
+		}
+		prev_ProcessUtility((Node *) alterstmt, queryString, context, params, dest, completionTag);
```

Comment:
So, my understanding of this is that the SPI upcall takes care of updating the metadata (in _timescaledb_catalog) while calling the original processUtility moves the actual data tables to the new schema? Just want to make sure I understand the need to call the original processUtility...